mayastor/src/subsys/config/pool.rs

```diff
@@ -0,0 +1,181 @@
+use std::{fmt::Display, fs, path::Path};
+
+use futures::lock::Mutex;
+use once_cell::sync::{Lazy, OnceCell};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    bdev::VerboseError,
+    core::{Cores, Reactor, Share},
+    lvs::Lvs,
+    pool::PoolsIter,
+    replica::{ReplicaIter, ShareType},
+};
+
+static CONFIG_FILE: OnceCell<String> = OnceCell::new();
+
+/// Initialise the config file location
+fn init_config_location<P>(file: P)
+where
+    P: AsRef<Path> + Display + ToString,
+{
+    CONFIG_FILE.get_or_init(|| file.to_string());
+}
+
+#[inline]
+/// Return the config file location
```

Comment:
Btw, usually the comments go above the "#[]" macros

Reply:
I think I agree, however did a quick search through our code and we aren't consistent at all.

Reply:
There is really no need to inline this at all.

Reply:
no longer inligned